new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleNginxValidation.java

```diff
@@ -0,0 +1,618 @@
+        adminServerPodName, domainNamespace);
+    checkServiceCreated(adminServerPodName);
+
+    // check managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check managed server service {0} is created in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkServiceCreated(managedServerPrefix + i);
+    }
+  }
+
+  @Test
+  @Order(2)
+  @DisplayName("Create an ingress for a WebLogic domain in the specified domain namespace")
+  public void testCreateIngress() {
+
+    // create an ingress in domain namespace
+    assertThat(assertDoesNotThrow(() -> createIngress(domainNamespace, domainUid)))
+            .as("createIngress succeeds")
+            .withFailMessage(String.format("failed to create an ingress for domain %s in namespace %s",
+                domainUid, domainNamespace))
+            .isTrue();
+
+    // check the ingress is created
+    String ingressName = domainUid + "-nginx";
+    assertThat(assertDoesNotThrow(() -> getIngressList(domainNamespace)))
+        .as(String.format("found the ingress %s in namespace %s", ingressName, domainNamespace))
+        .withFailMessage(String.format("can not find ingress %s in namespace %s", ingressName, domainNamespace))
+        .contains(ingressName);
+
+    logger.info("ingress is created in namespace {0}", domainNamespace);
+  }
+
+  @Test
+  @Order(3)
+  @DisplayName("Verify the application can be accessed through the ingress controller ")
+  public void testSampleAppThroughIngressController() throws Exception {
+
+    List<String> managedServerNames = new ArrayList<>();
+    for (int i = 1; i <= replicaCount; i++) {
+      managedServerNames.add(managedServerNameBase + i);
+    }
+
+    // check that Nginx can access the sample apps from all managed servers in the domain
+    String curlCmd = String.format("curl --silent --noproxy '*' -H 'host: %s' http://%s:%s/sample-war/index.jsp",
+        domainUid + ".org", K8S_NODEPORT_HOST, nodeportshttp);
+    assertThat(callWebAppAndCheckForServerNameInResponse(curlCmd, managedServerNames, 50))
+        .as("Nginx can access the sample app from all managed servers in the domain")
+        .withFailMessage("Nginx can not access the sample app from one or more of the managed servers")
+        .isTrue();
+  }
+
+  /**
+   * TODO: remove this after Sankar's PR is merged
+   * The cleanup framework does not uninstall Nginx release. Do it here for now.
+   */
+  @AfterAll
+  public void tearDownAll() {
+    // uninstall Nginx release
+    if (nginxHelmParams != null) {
+      assertThat(uninstallNginx(nginxHelmParams))
+          .as("Test uninstallNginx returns true")
+          .withFailMessage("uninstallNginx() did not return true")
+          .isTrue();
+    }
+  }
+
+  /**
+   * Install WebLogic operator and wait until the operator pod is ready.
+   */
+  private static void installAndVerifyOperator() {
```

Comment:
this feels like it might be universally reusable, maybe it belongs in a different class

Reply:
Vanaja will create a PR to re-organize the common methods used by everybody.

Reply:
Just to clarify, I am not working on re-organizing the common methods..I will be creating a basic domain image for MII in extension class which can be used by It classes.